### PR TITLE
Relative indirect pointer

### PR DESCRIPTION
**Pull request by slavapestov**

### CODE REVIEW

**Comment by slavapestov** [comment]
Hey @jckarter, does this look reasonable?

**Comment by jckarter** [comment]
Don't we already have `RelativeIndirectablePointer`?

File: lib/IRGen/ConstantBuilder.h [hunk 1]
```diff
@@ -94,6 +94,15 @@ class ConstantAggregateBuilderBase
...
   }
 
+  /// Add an indirect relative reference to the given address.
+  /// The target must be a "GOT-equivalent", i.e. a pointer to an
+  /// external object.
+  void addIndirectRelativeAddress(ConstantReference reference) {
+    assert(reference.isIndirect());
+    addRelativeOffset(IGM().RelativeAddressTy,
+                      reference.getValue());
+  }
```
Comment (jckarter) [inline lib/IRGen/ConstantBuilder.h]: There's already an overload of `addRelativeAddress` that takes a `ConstantReference` and does this.

**Comment by slavapestov** [comment]
@jckarter Sorry for the confusion, the second patch is unrelated. I need to use indirect pointers to refer to dispatch thunks in other modules.

**Pull request closed without merging**